### ISSUE TITLE
Flutter package publishing fails

### ISSUE THREAD

**Issue by mtrezza** [parse-community/Parse-SDK-Flutter#835]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest version of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Flutter SDK](https://github.com/parse-community/Parse-SDK-Flutter/releases). 

### Issue Description

Publishing the flutter package fails with error:

```
Package validation found the following potential issues:
* The name of "lib/parse_server_sdk.dart", "parse_server_sdk", should match the name of the package, "parse_server_sdk_flutter".
  This helps users know what library to import.
* line 8, column 1 of test/src/storage/core_store_directory_io_test.dart: This package does not have path_provider_platform_interface in the `dependencies` or `dev_dependencies` section of `pubspec.yaml`.
    ╷
  8 │ import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
* line 9, column 1 of test/src/storage/core_store_directory_io_test.dart: This package does not have plugin_platform_interface in the `dependencies` or `dev_dependencies` section of `pubspec.yaml`.
    ╷
  9 │ import 'package:plugin_platform_interface/plugin_platform_interface.dart';
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```
### Steps to reproduce

See https://github.com/parse-community/Parse-SDK-Flutter/actions/runs/4304405195/jobs/7505394193

### Actual Outcome

Package publish fails.

The CI passed in https://github.com/parse-community/Parse-SDK-Flutter/pull/826, so this is rather unexpected. Maybe we should add the publish dry-run to the PR CI to prevent his in the future?

### Expected Outcome

Package should publish.

**Comment by Nidal-Bakir** [comment]
@mtrezza 

can you try to add 
`path_provider_platform_interface: ^2.0.6`
in `dependencies` or `dev_dependencies` section of parse_server_sdk_flutter `pubspec.yaml

**Comment by mtrezza** [comment]
Thanks for the suggestion; would you mind opening a PR?

**Comment by mtrezza** [comment]
Still fails, see https://github.com/parse-community/Parse-SDK-Flutter/actions/runs/4338120170/jobs/7574582740

**Comment by Nidal-Bakir** [comment]
Fixed by #848 #846 #837

**Comment by mtrezza** [comment]
Thanks @Nidal-Bakir, well done!
